### ISSUE TITLE
CMakeLists file missing export

### ISSUE THREAD

**Issue by jeremyong** [cppformat/cppformat#257]
This would be nice to allow the cppformat project to be transitively included in an existing CMake build tree without requiring a separate make install step.

**Comment by vitaut** [comment]
It's already possible to include cppformat in a CMake project via `add_subdirectory`. Is there any advantage of using `export`?

**Comment by jeremyong** [comment]
Yes. Suppose I have library A which links cppformat and I want all users of library A to transitively link cppformat as well if they link A. Further, suppose I want A to be exported. The export will fail because A now publicly links cppformat which is not part of the export list. I'm not sure how to resolve this without directly modifying cppformat's CMakeLists.txt itself. It would be nice if an option was provided to specify an export list (optionally). My Cmake-fu is not 100% yet, but I believe this is the correct use case.

**Comment by vitaut** [comment]
It's possible to simply add cppformat with `add_subdirectory` and export `cppformat` or any other target:

```cmake
add_subdirectory(cppformat)
export(TARGETS cppformat FILE test.cmake)
```
Will it work for you?

**Comment by jeremyong** [comment]
I was referring to adding the cppformat target to an existing export list which isn't possible outside of the cppformat CMakeLists.txt file I believe. I'll play around with things some tomorrow and see if I can get something to work.

**Comment by vitaut** [comment]
It would help if you showed an example of what you are expecting to have in the cppformat's `CMakeFiles.txt` or, better, submitted a PR. I haven't used CMake export before and I'm not sure I fully understand what's the purpose of it other than for cross-compilation, but I'll be happy to accept the export-related changes if they simplify some use case.

**Comment by vitaut** [comment]
@jeremyong, have my suggestion worked for you or have you found an alternative?

**Comment by vitaut** [comment]
Fixed by #264.
